### PR TITLE
Make "sp-update-overlays" work from a clean render

### LINKED ISSUE #2813: [Bug]: Overlay.update() should release previously used measures before updating
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

Overlay

### Expected behavior

`Overlay.update()` should update not only the position of an overlay, but also it's available size, when restricted by the view port.

### Actual behavior

If the overlay has previously been sized, then the previous size will override the requirements of the view port.

### Screenshots

_No response_

### What browsers are you seeing the problem in?

_No response_

### How can we reproduce this issue?

1. Go to [here](https://studio.webcomponents.dev/edit/1YZCcOq4PMMLIJVekDBk/src/index.ts?branch=overlay%40wfVpXdsTYhf9mY2slxW4t2Y2SjC3&p=stories)
2. Click on "Picker"
3. See that `height` and `max-height` have been applied to the `active-overlay`.
4. Click on "More things"
5. See that content becomes available on scroll, but the overlay not resize to the available size of the view port.

### Sample code that illustrates the problem

https://studio.webcomponents.dev/edit/1YZCcOq4PMMLIJVekDBk/src/index.ts?branch=overlay%40wfVpXdsTYhf9mY2slxW4t2Y2SjC3&p=stories

### Logs taken while reproducing problem

_No response_

### PR DESCRIPTION
**Pull request by Westbrook**
## Description
When dispatching `sp-update-overlays` the expectation is the the size AND position of Overlays is updated. This change ensures that size rules applied as part of previous layout application passes is _removed_ before the update workflow is reentered.

## Related issue(s)
- fixes #2813

## How has this been tested?
-   [ ] _Test case 1_
    1. Go [here](https://sp-update-overlays--spectrum-web-components.netlify.app/storybook/?path=/story/overlay--accordion)
    2. Open the Overlay
    3. Toggle open Accordion Items until the overlay fills the vertical height of the viewport
    4. See that it is updated to fill the viewport but never overflows the viewport.
-  [ ] _Test case 2_
    1. Go [here](https://studio.webcomponents.dev/edit/1YZCcOq4PMMLIJVekDBk/src/index.ts?branch=overlay%40wfVpXdsTYhf9mY2slxW4t2Y2SjC3&p=stories)
    2. See how the size wasn't appropriately updated before.

## Types of changes
-   [x] Bug fix (non-breaking change which fixes an issue)

## Checklist
-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [ ] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [x] I have added tests to cover my changes.
-   [x] All new and existing tests passed.

### CODE REVIEW

**Review by najikahalsema — COMMENTED** [review]
It looks good to me, but should we be testing this in our test suite somehow? 🤔

**Comment by Westbrook** [comment]
@najikahalsema good call, added the equivalent of Test Case 1 as a unit test in 849a9dbe399

File: packages/overlay/src/ActiveOverlay.ts [hunk 1]
```diff
@@ -426,6 +426,16 @@ export class ActiveOverlay extends SpectrumElement {
         this.setOverlayPosition();
     };
 
+    public resetOverlayPosition = (): void => {
+        this.style.setProperty('max-height', '');
```
Comment (hunterloftis) [inline packages/overlay/src/ActiveOverlay.ts:430]: Are `style.setProperty('max-height', '')` and `style.removeProperty('max-height')` equivalent?

**Review by hunterloftis — DISMISSED** [review]

File: packages/overlay/src/ActiveOverlay.ts (hunk 1 above)
Comment (Westbrook) [reply packages/overlay/src/ActiveOverlay.ts:430]: Oooh, I've not used that API, it seems like it should be, so I'll test that as it seems like a nicer future to maintain.

**Pull request merged**